### PR TITLE
Add fix in docker test for Tumbleweed failure

### PR DESCRIPTION
**Pull request by punkioudi**
Fixes failure in docker test for TW failure.

- Related ticket: https://progress.opensuse.org/issues/67216)
- Needles: N/A
- Verification run: [SLES15sp1](http://10.161.229.247/tests/234#) [SLES15](http://10.161.229.247/tests/235#) [SLES12sp5](http://10.161.229.247/tests/236#) [SLES12sp4](http://10.161.229.247/tests/237#) [SLES12sp3](http://10.161.229.247/tests/238#)

openqa.suse.de : [SLES15sp1](https://openqa.suse.de/tests/4290989#details) [SLES15sp2](https://openqa.suse.de/tests/4291917#details) [SLES15sp2 JeOS](https://openqa.suse.de/tests/4291931#details)
openqa.opensuse.org : [Tumbleweed aarch64](https://openqa.opensuse.org/tests/1281317#) [Tumbleweed x86_64](https://openqa.opensuse.org/tests/1281423#)

### CODE REVIEW

File: lib/containers/docker.pm [hunk 1]
```diff
@@ -62,7 +63,13 @@ sub test_built_img() {
     assert_script_run("mkdir templates");
     assert_script_run("cd templates");
     assert_script_run("wget --quiet " . data_url('containers/index.html'));
-    assert_script_run("docker run -dit -p 8888:5000 -v ~/templates:\/usr/src/app/templates myapp https://openqa.suse.de//api/v1/jobs/${id}");
+    if (is_sle()) {
```
Comment (jlausuch) [inline lib/containers/docker.pm:66]: you can also do 1 liner 
`$path = is_sle() ? "openqa.suse.de" : "openqa.opensuse.org";`

However, I'm not sure I like this solution, we are still hardcoding 2 URLs and this makes things not really portable (e.g. I can't execute this test on my local or any other environment). I've seen other tests providing the url as an external variable, e.g. [boot_from_pxe.pm](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/boot/boot_from_pxe.pm#L62).

Maybe you can do something like:

```
my $path = is_sle() ? "openqa.suse.de" : "openqa.opensuse.org";`
$path = get_var('OPENQA_URL', $path);
```
or similar...

File: lib/containers/docker.pm [hunk 2]
```diff
@@ -27,6 +27,7 @@ our @EXPORT = qw(set_up get_vars build_img test_built_img);
 my $dir     = '';
 my $id      = '';
 my $version = '';
+my $path    = '';
```
Comment (jlausuch) [inline lib/containers/docker.pm:30]: you don't need this here, since it's used only in a single place.

File: lib/containers/docker.pm (hunk 1 above)
Comment (punkioudi) [reply lib/containers/docker.pm:66]: Thank you Jose for the feedback. The automatic variable OPENQA_URL, will return the localhost and not the public IP (as also other variables, like host_ip and OPENQA_HOSTNAME)
I haven't found a variable to return the public IP (eg when I execute locally, I need the IP of the machine with the openqa installed).
In any case, I will modify a little bit the test, but keep the logic that checks the network outside the container as it is.

Comment (jlausuch) [reply lib/containers/docker.pm:66]: Can you point me where you see it returning localhost?

I've seen some jobs in OSD and it returns `"OPENQA_URL": "openqa.suse.de",`
https://openqa.suse.de/tests/4285682/file/vars.json

Comment (punkioudi) [reply lib/containers/docker.pm:66]: For the openqa.suse.de it works, but when somebody wants to clone the test and run it locally, as I mentioned before, the OPENQA_URL will not return the public ip of the local openqa machine and the test will fail.

File: lib/containers/docker.pm [hunk 3]
```diff
@@ -56,8 +46,8 @@ sub build_img() {
 sub test_built_img() {
     assert_script_run("mkdir /root/templates");
     assert_script_run "curl -f -v " . data_url('containers/index.html') . " > /root/templates/index.html";
-    assert_script_run("docker run -dit -p 8888:5000 -v /root/templates:\/usr/src/app/templates myapp https://openqa.suse.de//api/v1/jobs/${id}");
+    assert_script_run("docker run -dit -p 8888:5000 -v ~/templates:\/usr/src/app/templates myapp www.google.com");
```
Comment (ggkioulis) [inline lib/containers/docker.pm:49]: Cool, this looks like a more robust solution.
I must say I liked the openQA api approach more but it seems that at some point, cloning the test to a local machine would give a headache inducing failure to someone.

File: tests/console/docker.pm [hunk 4]
```diff
@@ -53,6 +53,8 @@ sub test_seccomp {
 
 sub run {
     select_console("root-console");
+    #my $hostnam = autoinst_url();
+    #record_info("$hostnam");
```
Comment (jlausuch) [inline tests/console/docker.pm:57]: lefotvers?

File: data/containers/app.py [hunk 5]
```diff
@@ -19,7 +19,10 @@ def index():
         print("Program requires at least one argument!")
         sys.exit()
 
-    request = urllib.request.urlopen(url, context=ctx)
-    data = json.loads(request.read())
-    version = (data["job"]["settings"]["VERSION"])
+    cmd = "curl -I " + url
+    response = os.system(cmd)
+    if response == 0:
```
Comment (jlausuch) [inline data/containers/app.py:24]: `message = response == 0 ? 'pass' : 'not pass';`

File: lib/containers/docker.pm (hunk 3 above)
Comment (jlausuch) [reply lib/containers/docker.pm:49]: Yes, previous version was giving problems in local envs. This works as long as there is internet connectivity, which is for most cases.

**Pull request merged**